### ISSUE TITLE
Drop support for PHP 8.1 - 2026-02-01

### ISSUE THREAD

**Issue by justafish** [Lullabot/drainpipe#523]
PHP 8.1 reaches end-of-life on December 31, 2025 and should be removed from Drainpipe's supported versions. PHP 8.4 is now supported by Drupal 10.4+ and Drupal 11 and should be added.

  Changes Needed

  - Update composer.json PHP version constraint from ^8.1 to ^8.2
  - Update drainpipe-dev/composer.json PHP version constraint from ^8.1 to ^8.2
  - Add PHP 8.5 to CI test matrix in .github/workflows/ files
  - Remove PHP 8.1 from CI test matrix
  - Update any documentation mentioning PHP 8.1 support

  References

  - https://www.php.net/supported-versions.php
  - https://www.drupal.org/docs/getting-started/system-requirements/php-requirements

**Comment by mrdavidburns** [comment]
related: #473

**Comment by deviantintegral** [comment]
I don't think we should drop a version of PHP until it's actually EOL. Otherwise, incoming projects may be unable to adopt Drainpipe without first upgrading PHP, which can be a much bigger project.

What if we make it a practice (and we could document in the README) that we support PHP... 2 months after a version goes EOL?

**Comment by justafish** [comment]
:+1 I've added a due date of 1st January 2025 for this
